### PR TITLE
python311Packages.kornia: 0.7.1 -> 0.7.2

### PR DESCRIPTION
**Pull request by chpatrick**
## Description of changes

I'm not sure this is the best way to deal with the nesting in `kornia-rs`, so if there's a better way I'm happy to hear it. :)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/kornia-rs/default.nix [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  ];
+
+  cargoRoot = "py-kornia";
+  cargoDeps = rustPlatform.importCargoLock {
+    lockFile = ./Cargo.lock;
+  };
+
+  prePatch = ''
+    # The path dependency doesn't vendor the dependencies correctly, so get kornia-rs from crates instead.
+    patch -p1 <<END_PATCH
```
Comment (dotlambda) [inline pkgs/development/python-modules/kornia-rs/default.nix:36]: Please use `patches` and `substituteAll`.

File: pkgs/development/python-modules/kornia-rs/default.nix [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+, nasm
+}:
+
+buildPythonPackage rec {
+  pname = "kornia-rs";
+  version = "0.1.2";
+
+  src = fetchFromGitHub {
+    owner = "kornia";
+    repo = pname;
```
Comment (dotlambda) [inline pkgs/development/python-modules/kornia-rs/default.nix:15]: #277994

File: pkgs/development/python-modules/kornia-rs/default.nix [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    rev = "refs/tags/v${version}";
+    hash = "sha256-7toCMaHzFAzm6gThVLBxKLgQVgFJatdJseDlfdeS8RE=";
+  };
+
+  format = "pyproject";
```
Comment (dotlambda) [inline pkgs/development/python-modules/kornia-rs/default.nix:20]:
```suggestion
  pyproject = true;
```
Please move this below `version`.

File: pkgs/development/python-modules/kornia-rs/default.nix [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+    cp ${./Cargo.lock} py-kornia/Cargo.lock
+  '';
+
+  maturinBuildFlags = [ "-m" "py-kornia/Cargo.toml" ];
+
+  dontUseCmakeConfigure = true; # We only want to use CMake to build some Rust dependencies.
+
+  meta = with lib; {
+    homepage = "https://github.com/kornia/kornia-rs/tree/main";
```
Comment (dotlambda) [inline pkgs/development/python-modules/kornia-rs/default.nix:58]:
```suggestion
    homepage = "https://github.com/kornia/kornia-rs";
```

File: pkgs/development/python-modules/kornia-rs/default.nix [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+     [dependencies]
+     anyhow = "1.0.80"
+    -kornia-rs = { path = ".." }
+    +kornia-rs = { version = "${version}" }
+     pyo3 = { version = "0.20", features = ["extension-module"] }
+     numpy = { version = "0.20.0" }
+
+    END_PATCH
+
+    cp ${./Cargo.lock} py-kornia/Cargo.lock
```
Comment (dotlambda) [inline pkgs/development/python-modules/kornia-rs/default.nix:50]: Can't we patch the upstream one instead?

File: pkgs/development/python-modules/kornia/default.nix [hunk 6]
```diff
@@ -4,11 +4,12 @@
...
 , packaging
 , torch
+, kornia-rs
 }:
 
 buildPythonPackage rec {
   pname = "kornia";
-  version = "0.7.1";
+  version = "0.7.2";
   format = "pyproject";
```
Comment (dotlambda) [inline pkgs/development/python-modules/kornia/default.nix:13]:
```suggestion
  pyproject = true;
```

File: pkgs/development/python-modules/kornia/default.nix [hunk 7]
```diff
@@ -17,10 +18,11 @@ buildPythonPackage rec {
     owner = pname;
     repo = pname;
```
Comment (dotlambda) [inline pkgs/development/python-modules/kornia/default.nix:19]: #277994

**Review by dotlambda — COMMENTED** [review]
Please move the package addition to a separate commit and fix the commit message.
Also, `meta.changelog` would be nice.

File: pkgs/development/python-modules/kornia-rs/default.nix [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+
+  maturinBuildFlags = [ "-m" "py-kornia/Cargo.toml" ];
+
+  dontUseCmakeConfigure = true; # We only want to use CMake to build some Rust dependencies.
+
+  meta = with lib; {
+    homepage = "https://github.com/kornia/kornia-rs/tree/main";
+    description = "Python bindings to Low-level Computer Vision library in Rust";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ bcdarwin ];
```
Comment (dotlambda) [inline pkgs/development/python-modules/kornia-rs/default.nix:61]: This should be you unless @bcdarwin agreed to take over maintainership.

Comment (chpatrick) [reply pkgs/development/python-modules/kornia-rs/default.nix:61]: It's a helper package for `kornia` though, which he maintains.

File: pkgs/development/python-modules/kornia-rs/default.nix (hunk 5 above)
Comment (chpatrick) [reply pkgs/development/python-modules/kornia-rs/default.nix:50]: That one is different though because they use the path dependency.

**Comment by chpatrick** [comment]
@bcdarwin @dotlambda PTAL :)

**Comment by dotlambda** [comment]
Doesn't eval.

File: pkgs/development/python-modules/kornia-rs/default.nix (hunk 8 above)
Comment (dotlambda) [reply pkgs/development/python-modules/kornia-rs/default.nix:52]: Why is this still not you?

Comment (chpatrick) [reply pkgs/development/python-modules/kornia-rs/default.nix:52]: See the comment above, I didn't get an answer yet.

Comment (dotlambda) [reply pkgs/development/python-modules/kornia-rs/default.nix:52]: So why do you ask me to take a look while this isn't fixed yet?

File: pkgs/development/python-modules/kornia/default.nix (hunk 6 above)
Comment (dotlambda) [reply pkgs/development/python-modules/kornia/default.nix:13]: Still unfixed.

File: pkgs/development/python-modules/kornia-rs/default.nix (hunk 8 above)
Comment (chpatrick) [reply pkgs/development/python-modules/kornia-rs/default.nix:52]: I asked @bcdarwin too, and I wanted to know if you had any other issues.

**Comment by chpatrick** [comment]
@dotlambda I couldn't reach bcdarwin so I put myself as maintainer, do you have any other concerns or are we good to go?

**Comment by nh2** [comment]
@GrahamcOfBorg eval

**Comment by nh2** [comment]
This looks good to me, I'll merge it soon if there are no other requests for changes.

The

> Still unfixed.

comments look like they were retracted, I assume this is because they were meant for the newly added derivation (where they were addressed) instead of for the old derivation, which is unmodified (that look sensible to me for this PR).

**Pull request merged**